### PR TITLE
provider/azure: update azure-sdk-for-go.

### PR DESCRIPTION
**Pull request by mhilton**
Update the azure-sdk-for-go and the associated go-autorest
packages.  This makes the dependencies resolvable for any downstream
projects using go mod. It also allows the removal of homegrown SDK
implementations for APIs that weren't available in previous versions
of the SDK.

### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [ ] Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?
 - [ ] Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?
 - [x] Do comments answer the question of why design decisions were made?

----

## Description of change

Update the version of the Azure SDK linked to juju. The main motivation was to ensure that the azure SDK and the k8s clients were using a roughly similar version of go-autorest. This allows go.mod based builds to be able to resolve to a working module set. A fringe benefit is that it allows the removal of some customized azure interface code that is now handled by the SDK. 

## QA steps

```sh
# I created a new JUJU_DATA path so that existing credentials wouldn't be used.
$ JUJU_DATA=`mktemp -d juju-test.XXXXXX` export JUJU_DATA
$ juju add-credential azure
...
$ juju bootstrap --build-agent azure test-azure-001
$ juju deploy postgresql -n 2 --storage='pgdata=azure,30G'
# wait until everything has deployed without any problems.
$ juju destroy-controller test-azure-001 --destroy-all-models --destroy-storage
```

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: provider/azure/internal/errorutils/errors.go [hunk 1]
```diff
@@ -72,3 +78,70 @@ func hasDenialStatusCode(err error) bool {
...
+// CheckForGraphError attempts to unmarshal the body into a GraphError.
+// If this succeeds then the GraphError is returned as an error,
+// otherwise the response is passed on to the next Responder.
+func CheckForGraphError(r autorest.Responder) autorest.Responder {
+	return autorest.ResponderFunc(func(resp *http.Response) error {
+		if resp.Body != nil {
+			defer resp.Body.Close()
+			b, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return err
```
Comment (hpidcock) [inline provider/azure/internal/errorutils/errors.go]: errors.Trace

File: provider/azure/internal/azureauth/serviceprincipal.go [hunk 2]
```diff
@@ -356,7 +374,9 @@ func (c *ServicePrincipalCreator) createRoleAssignment(sdkCtx context.Context, p
 	if err != nil {
 		return errors.Annotate(err, "listing role definitions")
 	}
+	logger.Debugf("%+v", result)
```
[the remark this answers is not in the document]
Comment (mhilton) [reply provider/azure/internal/azureauth/serviceprincipal.go]: Sorry, I thought I'd removed all my debugging prints.

File: provider/azure/internal/errorutils/errors.go [hunk 3]
```diff
@@ -72,3 +78,70 @@ func hasDenialStatusCode(err error) bool {
 	}
 	return false
 }
+
+// CheckForGraphError attempts to unmarshal the body into a GraphError.
+// If this succeeds then the GraphError is returned as an error,
+// otherwise the response is passed on to the next Responder.
+func CheckForGraphError(r autorest.Responder) autorest.Responder {
```
[the remark this answers is not in the document]
Comment (mhilton) [reply provider/azure/internal/errorutils/errors.go]: done

File: provider/azure/internal/errorutils/errors.go (hunk 1 above)
Comment (mhilton) [reply provider/azure/internal/errorutils/errors.go]: done

**Comment by hpidcock** [comment]
!!build!! Timeout in unrelated test.
FAIL: firewaller_test.go:1181: InstanceModeSuite.TestRemoteRelationIngressFallbackToPublic

**Comment by hpidcock** [comment]
$$merge$$

**Comment by hpidcock** [comment]
$$merge$$

**Comment by mhilton** [comment]
$$merge$$

**Pull request merged**